sdk/index.ts

```diff
@@ -346,80 +377,200 @@ export class MidcontractProtocol {
+    const amount = parseUnits(input.amount.toString(), 1);
+    const amountToClaim = parseUnits(String(input.amountToClaim || 0), token.decimals);
+    const { totalDepositAmount } = await this.escrowDepositAmount(amount, this.account.address, input.feeConfig);
+    console.log(totalDepositAmount);
+    const status = input.status || Status.PENDING;
+    await this.tokenRequireBalance(account.address, totalDepositAmount, input.token);
+    await this.tokenRequireAllowance(account.address, totalDepositAmount, input.token, userEscrow);
+    // const gasPrice = ethers.parseUnits("1000", "gwei"); // Example gas price: 100 Gwei
+    // const { maxFeePerGas } = await this.getGasPrice();
+    try {
+      const data = await this.public.simulateContract({
+        address: userEscrow,
+        abi: escrow,
+        account,
+        args: [
+          input.contractorAddress,
+          token.address,
+          amount,
+          amountToClaim,
+          input.timeLock,
+          input.recipientData,
+          +input.feeConfig,
+          status,
+        ],
+        functionName: "deposit",
+        gasPrice: BigInt(30000),
+      });
+      const hash = await this.send(data.request);
+      const receipt = await this.getTransactionReceipt(hash, waitReceipt);
+      return { id: hash, status: receipt ? receipt.status : "pending" };
+    } catch (error) {
+      if (error instanceof ContractFunctionExecutionError) {
+        throw new SimulateError(error.shortMessage);
+      } else {
+        throw new CoreMidcontractProtocolError(JSON.stringify(error));
+      }
+    }
   }
 
-  async escrowSubmit(depositId: bigint, data: string, waitReceipt = true): Promise<TransactionId> {
+  async escrowSubmit(userEscrow: Address, data: string, waitReceipt = true): Promise<TransactionId> {
+    const contractId = await this.currentContractId(userEscrow);
+    const salt = ethers.keccak256(ethers.toUtf8Bytes("42"));
     const { request } = await this.public.simulateContract({
-      address: this.escrow,
+      address: userEscrow,
       abi: escrow,
       account: this.account,
-      args: [depositId, data],
+      args: [contractId, data, salt as Hash],
       functionName: "submit",
     });
     const hash = await this.send(request);
     const receipt = await this.getTransactionReceipt(hash, waitReceipt);
     return { id: hash, status: receipt ? receipt.status : "pending" };
   }
 
-  escrowRefill(depositId: bigint, value: number, waitReceipt = true): Promise<TransactionId> {
+  escrowRefill(depositId: bigint, value: number, userEscrow: Address, waitReceipt = true): Promise<TransactionId> {
     return this.escrowApprove(
       {
         depositId,
         valueAdditional: value,
       },
+      userEscrow,
       waitReceipt
     );
   }
 
-  async escrowApprove(input: ApproveInput, waitReceipt = true): Promise<TransactionId> {
+  async deployEscrow(adminAddress: Address, url?: string): Promise<Address> {
```

Comment:
Even taking into account a significant change in the architecture of contracts, we can do a higher-level implementation at the level of using the SDK. The purpose of the SDK is to abstract the client from all the details of the logic and architecture of the contracts and provide a higher-level interface. This will allow you to minimize actions on the front-end and back-end during the process of integrating a new version of the SDK.